internal/types/host/keepalived_host_collector_host.go

```diff
@@ -37,7 +37,14 @@ func NewKeepalivedHostCollectorHost(useJSON bool, pidPath string) *KeepalivedHos
 	if k.version, err = k.getKeepalivedVersion(); err != nil {
 		logrus.WithError(err).Warn("Version detection failed. Assuming it's the latest one.")
 	}
-
+	
+	if useJSON {
+		supported, err := isEnableJsonSupported()
+        	if err != nil {
+			logrus.WithError(err).Warn("Json support detection failed. Please check keepalivedJSON flag")
```

Comment:
You probably want to do an early return here, right?

Reply:
I'll modify it later.